src/libs/actions/Policy.js

```diff
@@ -344,6 +345,39 @@ function createPolicyExpenseChats(policyID, invitedEmailsToAccountIDs, betas) {
     return workspaceMembersChats;
 }
 
+/**
+ * Build personal details objects for a given list of logins and accountIDs
+ *
+ * @param {Array<string>} logins Array of user logins
+ * @param {Array<number>} accountIDs Array of user accountIDs
+ * @returns {Object} - object with optimisticData, successData and failureData (object of personal details objects)
+ */
+function buildPolicyPersonalDetails(logins, accountIDs) {
+    const policyPersonalDetails = {
+        optimisticData: {},
+        successData: {},
+        failureData: {},
+    };
+
+    _.map(logins, (login, index) => {
+        const accountID = accountIDs[index];
+
+        if (_.isEmpty(personalDetails[accountID])) {
+            policyPersonalDetails.optimisticData[accountID] = {
+                login,
+                accountID,
+                avatar: UserUtils.getDefaultAvatarURL(accountID),
+                displayName: login,
+            };
+
+            policyPersonalDetails.successData[accountID] = null;
```

Comment:
Won't this cause a flicker?
1. Optimistically set account
2. On success, remove account
3. On success (server response), add account (probably another account)

Reply:
Hmm, same thing goes for `policyMembers_` We will have redundant members. I think we can keep that outside the scope as well. Let's just make sure we have optimistic data for new non existing members.